src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3388,39 +3388,63 @@ internal virtual BoundExpressionStatement BindConstructorInitializer(PrimaryCons
+        private BoundNode BindConstructorBody(ConstructorDeclarationSyntax constructor, BindingDiagnosticBag diagnostics, bool includesFieldInitializers)
         {
-            if (constructor.Initializer == null && constructor.Body == null && constructor.ExpressionBody == null)
+            ConstructorInitializerSyntax initializer = constructor.Initializer;
+            if (initializer == null && constructor.Body == null && constructor.ExpressionBody == null)
             {
                 return null;
             }
 
             Binder bodyBinder = this.GetBinder(constructor);
             Debug.Assert(bodyBinder != null);
 
-            if (constructor.Initializer?.IsKind(SyntaxKind.ThisConstructorInitializer) != true &&
+            bool thisInitializer = initializer?.IsKind(SyntaxKind.ThisConstructorInitializer) == true;
+            if (!thisInitializer &&
                 ContainingType.GetMembersUnordered().OfType<SynthesizedRecordConstructor>().Any())
             {
                 var constructorSymbol = (MethodSymbol)this.ContainingMember();
                 if (!constructorSymbol.IsStatic &&
                     !SynthesizedRecordCopyCtor.IsCopyConstructor(constructorSymbol))
                 {
                     // Note: we check the constructor initializer of copy constructors elsewhere
-                    Error(diagnostics, ErrorCode.ERR_UnexpectedOrMissingConstructorInitializerInRecord, constructor.Initializer?.ThisOrBaseKeyword ?? constructor.Identifier);
+                    Error(diagnostics, ErrorCode.ERR_UnexpectedOrMissingConstructorInitializerInRecord, initializer?.ThisOrBaseKeyword ?? constructor.Identifier);
                 }
             }
 
+            bool skipInitializer = false;
+            if (includesFieldInitializers
+                && thisInitializer
+                && IsDefaultValueTypeConstructor(ContainingType, initializer))
+            {
+                // The `: this()` initializer is ignored when it is a default value type constructor
+                // and we need to include field initializers into the constructor.
+                skipInitializer = true;
+            }
+
             // Using BindStatement to bind block to make sure we are reusing results of partial binding in SemanticModel
             return new BoundConstructorMethodBody(constructor,
                                                   bodyBinder.GetDeclaredLocalsForScope(constructor),
-                                                  constructor.Initializer == null ? null : bodyBinder.BindConstructorInitializer(constructor.Initializer, diagnostics),
+                                                  skipInitializer ? new BoundNoOpStatement(constructor, NoOpStatementFlavor.Default)
+                                                      : initializer == null ? null
+                                                      : bodyBinder.BindConstructorInitializer(initializer, diagnostics),
```

Comment:
>: [](http://example.com/codeflow?start=54&length=1)

Should this be indented further, since it's associated with the nested conditional expression?

Reply:
I think that's how we've indented such nested conditionals before.

Reply:
If it is, then when I next stumble onto those cases I'll reformat for readability. The way this is formatted is very confusing.